src/lib/y2storage/boot_requirements_strategies/analyzer.rb

```diff
@@ -264,7 +282,18 @@ def device_for_boot
         boot_planned_dev || boot_filesystem || root_planned_dev || root_filesystem || nil
       end
 
+      # Partition table type of boot disk
+      #
+      # @return [PartitionTables::Type, nil] partition table type of boot disk or nil
+      #   if it doesn't have a partition table
       def boot_ptable_type
+        # FIXME
+        #
+        # This needs to be sorted out to support plain disks (without partition table).
+        # The commented line below is possibly the right thing but it's implicitly assmued
+        # that an empty disk will get a gpt elsewhere.
```

Comment:
Assuming this will get a GPT partition table was the right thing to do when `BootRequirementsChecker` was created, since it's original goal was to serve the Proposal. Reusing the checks for the Partitioner is something that came later.

In other words, in the scope of `BootRequirementsChecker#needed_partitions` (which answers the question "what additional partitions should the proposal create?") the historical behavior is right. If the boot/root disk has no partition table, it will get a GPT during the process for sure.

In the scope of `BootRequiremetsChecker#errors` or `BootRequirementsChecker#warnings` (which answer the question "what is wrong with this layout?") this is wrong.